src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,295 @@
+			prototype._refresh = function () {
+				var self = this;
+
+				self._setValue(self.options.value);
+			}
+
+			prototype._setValue = function (value) {
+				var self = this,
+					options = self.options;
+
+				options.pastSelection = value;
+			}
+
+			prototype._build = function (element) {
+				var controllerElement = document.createElement("div"),
+					leftArrowElement = document.createElement("div"),
+					rightArrowElement = document.createElement("div"),
+					viewChangeElement = document.createElement("div"),
+					viewTableElement = document.createElement("table"),
+					spaceElement = document.createElement("tr"),
+					oneWeekElement = document.createElement("tr"),
+					monElement = document.createElement("td"),
+					tueElement = document.createElement("td"),
+					wedElement = document.createElement("td"),
+					thuElement = document.createElement("td"),
+					friElement = document.createElement("td"),
+					satElement = document.createElement("td"),
+					sunElement = document.createElement("td");
+
+				// Controller
+				controllerElement.classList.add("controller");
+				leftArrowElement.classList.add("calendarArrowLeft");
+				rightArrowElement.classList.add("calendarArrowRight");
+				viewChangeElement.classList.add("calendarSwitch");
+
+				viewChangeElement.id = "yearMonth";
+
+				controllerElement.appendChild(leftArrowElement);
+				controllerElement.appendChild(rightArrowElement);
+				controllerElement.appendChild(viewChangeElement);
+
+				// View Container
+				viewTableElement.classList.add("calendarView");
+				spaceElement.classList.add("topSpace");
+				oneWeekElement.classList.add("oneWeek");
+
+				viewTableElement.id = "calendar";
+
+				monElement.innerHTML = weeks[0];
+				oneWeekElement.appendChild(monElement);
+				tueElement.innerHTML = weeks[1];
+				oneWeekElement.appendChild(tueElement);
+				wedElement.innerHTML = weeks[2];
+				oneWeekElement.appendChild(wedElement);
+				thuElement.innerHTML = weeks[3];
+				oneWeekElement.appendChild(thuElement);
+				friElement.innerHTML = weeks[4];
+				oneWeekElement.appendChild(friElement);
+				satElement.innerHTML = weeks[5];
+				oneWeekElement.appendChild(satElement);
+				sunElement.innerHTML = weeks[6];
+				sunElement.id = "sunday";
+				oneWeekElement.appendChild(sunElement);
+
+				viewTableElement.appendChild(spaceElement);
+				viewTableElement.appendChild(oneWeekElement);
+
+				// append Elements
+				element.appendChild(controllerElement);
+				element.appendChild(viewTableElement);
+
+				return element;
+			};
+
+			prototype._destroy = function () {
+				var self = this;
+
+				self.options = null;
+				self.unBindEvents(self.element);
+			};
+
+			ns.widget.mobile.Calendar = Calendar;
+			engine.defineWidget(
+				"Calendar",
+				".ui-calendar",
+				[],
+				Calendar,
+				"mobile"
+			);
+			//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+			return ns.widget.mobile.Calendar;
+		}
+	);
+	//>>excludeEnd("tauBuildExclude");
```

Comment:
ditto.

Reply:
> ditto.
This comment is needed in build process.

Reply:
ditto.